Allow configuration via environment variables

hannahbast opened the pull request:
The `qlever` script can now get its config either from a `Qleverfile` (just like before) *or* from environment variables (this is new).  The environment variables have names that are fully analogous to the variable names in the `Qleverfile`. For example, the variable `PORT` in section `server` of the `Qleverfile` corresponds to the environment variable `QLEVER_SERVER_PORT`.

The `qlever` script now checks if *either* a `Qleverfile` exists *or* if environment variables of the form `QLEVER_SECTION_VARIABLE`  exist. If exactly one of these exists, it will automatically parse the configuration from there. If both exist, it will complain (it would be easy to merge both, by letting one override the other, but I don't see a use case for that, but the potential for unexpected behavior instead, since a user might not be aware of which environment variables are set).

There is now also a new command `qlever config` with various convenience options, in particular: `--show-qleverfile`, `--show-envvars`, `--set-envvars-from-qleverfile`, and `--unset-envvars`. See `qlever config --help` for a description of what exactly they do.

Commit:
Allow configuration via environment variables

hannahbast commented:
@ludovicm67 Can you please have a look at this and try it out and let me know what you think? I think it nicely addresses the problem we talked about last week. For testing, in oder to get the environment variables for one of the preconfigured Qleverfiles, you can do the following:

```
qlever config --get-qleverfile olympics
qlever config --set-envvars-from-qleverfile
```

ludovicm67 commented:
@hannahbast I will give it a try!

This is similar than what I've done here in a shell script: https://github.com/zazukoians/qlever-tests/blob/76068584b86a620ab1fdde4a309d81bdda733e36/docker/common/generate-qleverfile.sh

It's great that we are fully on the same page for the variable names 👍

Commit:
Fix minor bug

ludovicm67 reviewed (commented):
I gave it a try, and it seems that `qlever` is able to get its configuration from the environment variables successfully 👍 

Thanks a lot for this very useful change!

Qup42 commented on src/qlever/commands/config.py:
```diff
@@ -0,0 +1,196 @@
+        return {}
+
+    def additional_arguments(self, subparser) -> None:
+        subparser.add_argument(
+                "--get-qleverfile", type=str,
+                choices=self.qleverfile_names,
+                help="Get one the many pre-configured Qleverfiles")
+        subparser.add_argument(
+                "--show-qleverfile", action="store_true",
+                default=False,
+                help="Show the configuration from the Qleverfile "
+                "(if it exists)")
+        subparser.add_argument(
+                "--show-envvars", action="store_true", default=False,
+                help="Show all existing environment variables of the form "
+                "QLEVER_SECTION_VARIABLE")
+        subparser.add_argument(
+                "--varname-width", type=int, default=25,
+                help="Width for variable names in the output")
+        subparser.add_argument(
+                "--set-envvars-from-qleverfile", action="store_true",
+                default=False,
+                help="Set the environment variables that correspond to the "
+                "Qleverfile configuration (for copying and pasting)")
+        subparser.add_argument(
+                "--unset-envvars", action="store_true", default=False,
+                help="Unset all environment variables of the form "
+                "QLEVER_SECTION_VARIABLE (for copying and pasting)")
```
For me many of these feel more natural as a subcommand than as a parameter

Commit:
Merge remote-tracking branch 'origin/main' into add-config-command

Commit:
Split between commands `setup-config` and `show-config'

The options `--set-envvars-from-qleverfile` and `--unset-envvars` are
now part of the `setup-config` command (because they are about setting
up the configuration).

For showing the current configuration, a new command `show-config` has
been added.

Whether running `qlever` via environment variables works is now also
tested as part of the `qleverfiles-check` workflow.
